compiler/ast/dag/operator.go

```diff
@@ -150,14 +150,10 @@ type (
 	}
 
 	Pushdown struct {
-		Scan  Op               `json:"scan"`
-		Index []IndexPredicate `json:"index"`
+		Scan  Op      `json:"scan"`
+		Index *Filter `json:"index"`
```

Comment:
This feels to me like it wants to be called IndexFilter/index_filter (but I'm OK with the current name if you don't agree).

Reply:
Would you also change Scan to ScanFilter?